### ISSUE TITLE
Adjust 'family' description in OpenColorIO.h to match documentation

### ISSUE THREAD

**Issue by doug-walker** [AcademySoftwareFoundation/OpenColorIO#1223]
In the documentation (both v1 and v2), the family property of a color space is defined as:
"Allows for logical grouping of colorspaces within a UI."

And indeed, that is how it seems to be used in practice.  For example, in the ACES config, the color spaces {ACES2065-1, ACEScc, ACEScct, ACEScg} are all part of a family "ACES" and all of the various ARRI IDTs are in a family called "Input/ARRI".

However, in the OpenColorIO.h header file there are several places where 'family' has a different description.  For example line 1373: 
```
// *ColorSpaces* are specific to a particular image precision (float32,
// uint8, etc.), and the set of ColorSpaces that provide equivalent mappings
// (at different precisions) are referred to as a 'family'.
```
The proposed correction is to remove that sentence.

And line 1399:
```
     * Get the ColorSpace group name (used for equality comparisons)
     * This allows no-op transforms between different colorspaces.
     * If an equalityGroup is not defined (an empty string), it will be considered
     * unique (i.e., it will not compare as equal to other ColorSpaces with an
     * empty equality group).  This is often, though not always, set to the
     * same value as 'family'.
```
The proposed correction is to remove the last sentence ("This is often, ...").

And line 1988:
```
     *    This may provide higher fidelity than anticipated due to internal
     *    optimizations. For example, if the inputColorSpace and the
     *    outputColorSpace are members of the same family, no conversion
     *    will be applied, even though strictly speaking quantization
     *    should be added.
```
Note that this latter usage is definitely not consistent with the code -- the family property has no impact on the result of color processing.  The proposed change in this case is to replace 'family' with 'equalitygroup' in that sentence.

It seems the definition of 'family' changed during the development of v1 but the header file was not updated.  This should now be done as it could easily cause confusion among developers.

**Comment by Ansh-Sarkar** [comment]
@doug-walker could you pls assign this issue to me. Am relatively new to open source. Would love to start with this.

**Comment by doug-walker** [comment]
Done.  Thank you @Ansh-Sarkar.

**Comment by Ansh-Sarkar** [comment]
Thank you so much. Will submit a PR with the necessary changes as soon as possible

**Comment by Ansh-Sarkar** [comment]
@doug-walker have made the necessary changes and modified comments at required places. please review my P.R. #1227 .

**Comment by Ansh-Sarkar** [comment]
@doug-walker I have made the necessary changes. The CLA has been signed and all the tests have been passed. Please review my P.R. #1230

**Comment by doug-walker** [comment]
This issue is still open.  It's a simple one, so a good option for first-time contributors.

**Comment by akhilsahuji** [comment]
I am facing  error am I doing something wrong while  running below commands ?

 > ```
> cmake <path to OCIO root dir>/. -DOCIO_BUILD_DOCS=ON -DOCIO_BUILD_FROZEN_DOCS=ON
> make docs -j<thread count>
> ```

Errors
```
CMake Error at CMakeLists.txt:30 (project):
  Running

   'nmake' '-?'

  failed with:

   The system cannot find the file specified


CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
See also "C:/Users/akhil/Documents/GSoc/OpenColorIO/CMakeFiles/CMakeOutput.log".
```

**Comment by doug-walker** [comment]
Implemented via PR #1635.
